samples/affinidi-nextjs-nextauthjs/src/lib/clients/iota.ts

```diff
@@ -2,27 +2,90 @@ import {
   Configuration,
   ConfigurationsApi,
   PexQueryApi,
-} from "@affinidi-tdk/iota-client";
-import { getAuthProvider } from "./auth-provider";
+  IotaApi,
+  CallbackApi,
+  InitiateDataSharingRequestOKData,
+  FetchIOTAVPResponseOK,
+} from '@affinidi-tdk/iota-client'
+import { getAuthProvider } from './auth-provider'
+import { randomUUID } from 'crypto'
+import * as jose from 'jose'
 
 export async function listIotaConfigurations() {
-  const authProvider = getAuthProvider();
+  const authProvider = getAuthProvider()
   const api = new ConfigurationsApi(
     new Configuration({
       apiKey: authProvider.fetchProjectScopedToken.bind(authProvider),
     }),
-  );
-  const { data } = await api.listIotaConfigurations();
-  return data.configurations;
+  )
+  const { data } = await api.listIotaConfigurations()
+  return data.configurations
 }
 
 export async function listPexQueriesByConfigurationId(configurationId: string) {
-  const authProvider = getAuthProvider();
+  const authProvider = getAuthProvider()
   const api = new PexQueryApi(
     new Configuration({
       apiKey: authProvider.fetchProjectScopedToken.bind(authProvider),
     }),
-  );
-  const { data } = await api.listPexQueries(configurationId);
-  return data.pexQueries;
+  )
+  const { data } = await api.listPexQueries(configurationId)
+  return data.pexQueries
 }
+
+// NOTE: this is all-in-one method to test the flow is working
+export async function startIotaRedirectFlow() {
+  // TODO: get data from user, FE
+  const iotaConfigId =  process.env.IOTA_CONFIG_ID!
+  const queryId =  process.env.QUERY_ID!
+  const did =  process.env.DID!
+  const redirectUri =  process.env.REDIRECT_URI!
+
+  const authProvider = getAuthProvider()
+  const { iotaJwt, iotaSessionId } = await authProvider.createIotaToken(iotaConfigId, did)
+  // console.log({ iotaJwt })
+
+  // NOTE: for tdk-client AFFINIDI_TDK_ENVIRONMEN=dev will NOT work, we have to pass the basePath for dev environment
```

Comment:
FYI we also have a `with-env` branch that has the API GW stuff set up as well

https://github.com/affinidi/reference-app-affinidi-vault/tree/with-env